### DISCUSSION TITLE
SQS responds with an error, but no catch is executed

### DISCUSSION THREAD

**Discussion by adrian-goe** [aws/aws-sdk-js#4092] [General]
Confirm by changing [ ] to [x] below:
- [x] I've gone through [Developer Guide](https://docs.aws.amazon.com/sdk-for-javascript/v2/developer-guide/welcome.html) and [API reference](https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/)
- [x] I've checked [AWS Forums](https://forums.aws.amazon.com) and [StackOverflow](https://stackoverflow.com/questions/tagged/aws-sdk-js) for answers
- [x] This is an issue with version 2.x of the SDK

**Describe the question**
Hi, wie tried to send messages to SQS. In our setup, the max_message_size was limited to 2kb, without someone remembering it. Removing the limit fixed our bug. But finding this, was a struggle with some the following question. 

So we tried to send messages SQS:
```ts
return await sqsClient
      .send(command)
      .then(() =>
        console.info(`Sended x jobs to SQS`)
      )
      .catch((error) => console.error('some error...', error));
```

We were wondering, why it says, that messages were send, but dosn't apper in sqs.

After some unsuccessful debugging, we change the `then()` this:
```
.then((value) => console.info(`Sended x jobs to SQS`, JSON.stringify(value)))
```

The results are a bit surprising.

```json
{
    "$metadata": {
        "httpStatusCode": 200,
        "requestId": "...",
        "attempts": 1,
        "totalRetryDelay": 0
    },
    "Failed": [
        {
            "Id": "...",
            "SenderFault": true,
            "Code": "InvalidParameterValue",
            "Message": "One or more parameters cannot be validated. Reason: Message must be shorter than 2048 bytes."
        }
    ]
}
```

The httpCode is 200 but it's a failing request. My expectation would be, that this failed request would result in an error, that goes to the catch blog. 
Is there a reason for this behavior? It feels wrong.

We use `"@aws-sdk/client-sqs": "3.72.0"`

**Comment by yenfryherrerafeliz** [comment]
Hi @adrian-goe, thank you for reaching out. In order to attempt to reproduce the reported issue, could you please provide the following:
- The definition/values of the variable "command". 
    - Please redact any sensitive information!

<summary>The sample code below worked just fine for me:</summary>

<details>

```js
const sqsClient = new SQSClient({region: 'REGION'});

        // Set the parameters
        const params = {
            DelaySeconds: 10,
            MessageAttributes: {
                Title: {
                    DataType: "String",
                    StringValue: "Testing Message",
                },
                Author: {
                    DataType: "String",
                    StringValue: "Yenfry Herrera",
                },
                WeeksOn: {
                    DataType: "Number",
                    StringValue: "6",
                },
            },
            MessageBody: "A TEXT OVER THE LIMIT(2KB)",
            QueueUrl: "https://sqs.{REGION}.amazonaws.com/{ACCOUNT_ID}/{QUEUE-NAME}"
        };

        await sqsClient.send(new SendMessageCommand(params))
            .then((value)=>{console.log('Value: ', value);})
            .catch((reason)=>{console.log('Error: ', reason)});

```
</details>

I will be also moving this issue to the right [repo](https://github.com/aws/aws-sdk-js-v3/)!

Thank you!
